src/ga/ga_simple.rs

```diff
@@ -98,18 +98,20 @@ impl<T: GAIndividual + Clone> GeneticAlgorithm<T> for SimpleGeneticAlgorithm <T>
             new_individuals.push(new_ind);
         }
 
+        let best_old_individual = self.population.best().clone();
+
         // Evaluate the new population
-        // self.population.swap(new_individuals);
+        self.population.swap(new_individuals);
```

Comment:
This is a little bit in conflict with what I had devised for population statistics. Since the idea is to make our implementation highly concurrent, I tried to avoid a scenario in which a population and its statistics were out of sync (e.g. the population is at generation 7, but their statistics are still at generation 6). So a given population generation would either have no statistics yet or the same throughout its whole lifetime; this is reflected [here](https://github.com/mrjackinc/rust-monster/pull/22/files#diff-fe8a50b8d3f7fd3cef8b276a0f04e493R228): If no stats have been computed, they are computed and returned; if they have been computed before, they are simply returned -- they always correspond.

Each step of the GA would create a new `GAPopulation` (and replace its population member), ready to be statted. The `GAPopulation` object being replaced (current generation - 1) could then be archived (or its stats at least) in the [global statistics object](https://github.com/mrjackinc/rust-monster/pull/22/files#diff-78fc5807638a16cf09cdff0a3d70551dR86) with a simply move, rather than a copy. Allocating a new `GAPopulation` struct at every step of the GA would not be much more expensive than the current implementation: The bulk is in the individuals vector, which we are replacing here anyway.